### PR TITLE
The task has done

### PR DESCRIPTION
**Pull request by NekitCool1**

### CODE REVIEW

File: src/main/java/book/shop/dto/BookDto.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package book.shop.dto;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/dto/BookDto.java:4]:
```suggestion
```

File: src/main/java/book/shop/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package book.shop.dto;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class CreateBookRequestDto {
+
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/dto/CreateBookRequestDto.java]:
```suggestion
```

File: src/main/java/book/shop/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -1,24 +1,42 @@
...
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("XXX" + new Random().nextInt());
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/service/impl/BookServiceImpl.java]: it's better to remove for now, in future tasks you can add validation pattern or generaye isbn method

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/book/shop/dto/BookDto.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    private String title;
+
+    private String author;
+
+    private BigDecimal price;
+
+    private String description;
+
+    private String coverImage;
+}
```
Comment (MentorMateAcademy) [inline src/main/java/book/shop/dto/BookDto.java]: clear empty lines between

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/book/shop/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/main/java/book/shop/dto/CreateBookRequestDto.java]: not fixed

File: src/main/java/book/shop/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -41,13 +43,22 @@ public Book save(Book book) {
         return book;
     }
 
+    @Override
+    public Optional<Book> getBookById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(Book.class, id));
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find the book with id: " + id);
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't find the book with id: " + id);
```

File: src/main/java/book/shop/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -41,13 +43,22 @@ public Book save(Book book) {
...
+    }
+
     @Override
     public List<Book> findAll() {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery(
                     "SELECT b FROM Book b", Book.class).getResultList();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't ger all books from DB", e);
+            throw new EntityNotFoundException("Can't ger all books from DB " + e);
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all books from DB " + e);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/book/shop/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package book.shop.dto;
+
+import java.math.BigDecimal;
+
+public class CreateBookRequestDto {
+
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/dto/CreateBookRequestDto.java]: Remove empty line after class declaration
Also, remove empty lines between fields

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

**Review by Elena-Bruyako — APPROVED** [review]
good job

**Pull request merged**